rescompy/rescompy.py

```diff
@@ -740,48 +860,60 @@ def load(
             
         return result
 
-    def _get_states(
+    def get_states(
         self, 
         initial_state:      np.ndarray, 
-        inputs:             np.ndarray):
-        states = initial_state[None].repeat(inputs.shape[0] + 1,
-                                                   axis=0)
+        inputs:             np.ndarray,
+        ):
+        
+        states = initial_state[None].repeat(inputs.shape[0] + 1, axis=0)
 
         # Propagate the reservoir state with the input signal.
         return _get_states_driven(inputs, states,
                                          self.A.data, self.A.indices,
                                          self.A.indptr, self.A.shape,
                                          self.B, self.C,
                                          self.leaking_rate)
-    
+
     @validate_arguments(config=dict(arbitrary_types_allowed=True))
-    def train(
+    def train_saving_memory(
```

Comment:
why is this renamed?

Reply:
I don't think it's renamed. It is a new function. I think it just appears to git that it has been renamed because train_saving_memory() appears earlier in the code than does train().